src/sentry/ingest/transaction_clusterer/rules.py

```diff
@@ -152,6 +163,22 @@ def get_sorted_rules(project: Project) -> List[Tuple[ReplacementRule, int]]:
 
 
 def update_rules(project: Project, new_rules: Sequence[ReplacementRule]) -> None:
+    # Run the updates even if there aren't any new rules, to get all the stores
+    # up-to-date.
```

Comment:
Good question. This is required to update project options with the `last_seen` values in Redis, and there are no rules. This may happen in two situations:

1. Rules are no longer generated because the volume is low. This is not the case we should really care about, since the volume would be low enough to not have a significant impact on the infrastructure.
2. No new rules are created. It could be possible no new rules are created because all transaction names are perfectly parameterized; and even if the volume of transactions is very high, there are no rules to "update". In the current state, Redis and Project Options won't ever be merged, causing TTLs not to be updated and losing rules over time.

This is basically to cover the second case. There may be other edge cases I haven't considered too, even potential regressions on the clustering algorithm. I want to be safe here and keep project options always up-to-date. This shouldn't be problematic wrt writing to Postgres, since it only happens once per task run. I'll add a comment to keep this in mind in the future.

Reply:
> In the current state, Redis and Project Options won't ever be merged, causing TTLs not to be updated and losing rules over time.

Oh yeah, you are right of course!